rust-mode-tests.el

```diff
@@ -2024,6 +2024,172 @@ fn main() {
    '("r#\"\"\"#" font-lock-string-face
      "'q'" font-lock-string-face)))
 
+(ert-deftest rust-macro-font-lock ()
+  (rust-test-font-lock
+   "foo!\(\);"
+   '("foo!" font-lock-preprocessor-face))
+  (rust-test-font-lock
+   "foo!{};"
+   '("foo!" font-lock-preprocessor-face))
+  (rust-test-font-lock
+   "foo![];"
+   '("foo!" font-lock-preprocessor-face)))
+
+(ert-deftest rust-string-interpolation-matcher-works ()
+  (dolist (test '(("print!\(\"\"\)" 9 11 nil)
+                  ("print!\(\"abcd\"\)" 9 15 nil)
+                  ("print!\(\"abcd {{}}\"\);" 9 19 nil)
+                  ("print!\(\"abcd {{\"\);" 9 18 nil)
+                  ("print!\(\"abcd {}\"\);" 9 18 ("{}"))
+                  ("print!\(\"abcd {{{}\"\);" 9 20 ("{}"))
+                  ("print!\(\"abcd {}}}\"\);" 9 20 ("{}"))
+                  ("print!\(\"abcd {{{}}}\"\);" 9 20 ("{}"))
+                  ("print!\(\"abcd {0}\"\);" 9 18 ("{0}"))
+                  ("print!\(\"abcd {0} efgh\"\);" 9 23 ("{0}"))
+                  ("print!\(\"{1} abcd {0} efgh\"\);" 9 27 ("{1}" "{0}"))))
+    (destructuring-bind (text cursor limit matches) test
+      (with-temp-buffer
+        (insert text)
+        (goto-char cursor)
+        (dotimes (n (length matches))
+          (rust-string-interpolation-matcher limit)
+          ;; skip match 0
+          (should (equal (match-string 0) (elt matches n))))))))
+
+(ert-deftest rust-formatting-macro-font-lock ()
+  ; test that the block delimiters aren't highlighted and the comment
+  ; is ignored
```

Comment:
In elisp, comments that indent with the code start with `;;`.  A single semi is reserved for a comment to the right of code.